tests/dispersytestclass.py

```diff
@@ -63,24 +71,43 @@ def tearDown(self):
             self._logger.warning("Failing")
         assert not pending, "The reactor was not clean after shutting down all dispersy instances."
 
-    def create_nodes(self, amount=1, store_identity=True, tunnel=False, communityclass=DebugCommunity, autoload_discovery=False):
+    def create_nodes(self, amount=1, store_identity=True, tunnel=False, community_class=DebugCommunity,
+                     autoload_discovery=False, memory_database=True):
+        """
+        Creates dispersy nodes running a community.
+        :param amount: The amount of nodes that need to be created.
+        :param store_identity: If the identity is send to the central node.
+        :param tunnel: If the nodes is behind a tunnel or not
+        :param community_class: The class that the node will autoload.
+        :param autoload_discovery: If the discovery community is autoloaded.
+        :param memory_database: If a memory database is used.
+        :return: [(DebugNode)]
+        """
+
+        """ Override this method in a subclass with a different community class to test communities. """
         @inlineCallbacks
-        def _create_nodes(amount, store_identity, tunnel, communityclass, autoload_discovery):
+        def _create_nodes(amount, store_identity, tunnel, communityclass, autoload_discovery, memory_database):
             nodes = []
             for _ in range(amount):
                 # TODO(emilon): do the log observer stuff instead
                 # callback.attach_exception_handler(self.on_callback_exception)
+                memory_database_argument = {'database_filename': u":memory:"} if memory_database else {}
+                working_directory = unicode(mkdtemp(suffix="_dispersy_test_session"))
 
-                dispersy = Dispersy(ManualEnpoint(0), u".", u":memory:")
+                dispersy = Dispersy(ManualEnpoint(0), working_directory, **memory_database_argument)
                 dispersy.start(autoload_discovery=autoload_discovery)
 
                 self.dispersy_objects.append(dispersy)
 
-                node = DebugNode(self, dispersy, communityclass, c_master_member=self._mm)
+                node = self._create_node(dispersy, communityclass, self._mm)
                 yield node.init_my_member(tunnel=tunnel, store_identity=store_identity)
 
                 nodes.append(node)
             self._logger.debug("create_nodes, nodes created: %s", nodes)
             returnValue(nodes)
 
-        return blockingCallFromThread(reactor, _create_nodes, amount, store_identity, tunnel, communityclass, autoload_discovery)
+        return blockingCallFromThread(reactor, _create_nodes, amount, store_identity, tunnel, community_class,
+                                      autoload_discovery, memory_database)
+
+    def _create_node(self, dispersy, community_class, c_master_member):
```

Comment:
I used it somewhere else.

I used it in my own tests to setup a DebugNode with a specific curve by overriding the method.